### ISSUE TITLE
NOD | Prototype: Downloadable Copy of an Appeal

### ISSUE THREAD

**Issue by eileen-coforma** [department-of-veterans-affairs/va.gov-team#65936]
## Value Statement
**_As a_**  Veteran using a screenreader
**_I want to_** review a prototype of the "downloadable copy of an appeal" feature
**_So that_** I can accurately test the experience and value with assistive-technology users.

---
## Acceptance Criteria
- [ ] User can access the code prototype without having to authenticate
- [ ] The confirmation page has updated copy and link
- [ ] User has the option to click a primary link that leads to an HTML page with completed claim info
- [ ] User have the option to download a PDF version of the submission within 1 week (date displayed as a variable of submission date+7)
- [ ] User can read mock data for completed claim displayed on the page
- [ ] User has the option to click a button to print/save the page the browser's print page modal
- [ ] Code prototype includes accessibility considerations (properly coded headers)

## Designs and Build Notes
DESIGN: 
[Confirmation page](https://www.figma.com/file/BplQkEDZzD8NPPNmLrhvXv/Request-a-Board-Appeal-%2F-Notice-of-Disagreement-(VA-10182)?type=design&node-id=0-8647&mode=design&t=5lejqiCk08PHsjTK-11)
[Your Submitted Board Appeal page](https://www.figma.com/file/BplQkEDZzD8NPPNmLrhvXv/Request-a-Board-Appeal-%2F-Notice-of-Disagreement-(VA-10182)?type=design&node-id=0-8647&mode=design&t=5lejqiCk08PHsjTK-11)

```mermaid
flowchart TB;
   id1["Confirmation with link to completed claim"]-->id2["completed 
claim information HTML with Download PDF link"]-->id3[PDF ];
```

## Out of scope
- Backend code to retrieve data
- Email confirmation delivery
- A way to revisit completed form page

## Tasks
TBD

## Definition of Ready 
- [ ] Design has been decided and approved by team & enablement
- [ ] Clear value description
- [ ] Testable acceptance criteria
- [ ] Accessibility added to acceptance criteria
- [ ] Approved designs attached
- [ ] Sample data provided where appropriate
- [ ] Estimated to fit within the sprint
- [ ] Dependencies and blockers linked

## Definition of Done
- [ ] Meets acceptance criteria
- [ ] Code reviewed (internal)
- [ ] Submitted to staging
- [ ] Reviewed and approved by product and/or design

**Comment by saderagsdale** [comment]
Action item: Kyle and Robin to look into changes from LH then apply estimate @Mottie @kylesoskin

**Comment by Mottie** [comment]
Issues that need to be discussed:
- Since we're calling an API and not pointing to an existing file, the download should be a button, and not a link
- Should we include the confirmation page in the PDF? Doing this would require creating a template that the backend would have to fill out and prepend it to the PDF from Lighthouse. Who keeps the confirmation page PDF template up to date?
- Lighthouse's download endpoint will work:
  - _After_ the submission gets to a "submitted" state. Immediately after submitting, the confirmation page gets a "pending" status. I have no idea how long it takes to go from "pending" to "submitted"
  - The PDF is only available for 1 week after the form status gets to a "completed" state
- If the download _is_ immediately available when the confirmation page is available, what error message do we show the Veteran if the download fails? And what error message should we log to analytics?
- If the Veteran accidentally closes the confirmation page, how would they download the PDF?
  - I think we should collaborate with the Claim Status Tool team and include the link there...
  - Or, better yet, have our backend wait until the "submitted" status is available, then download & save the PDF to the Veteran's eFolder. I'm not sure why Lighthouse isn't doing this directly?
  - Maybe we could send the Veteran an email that includes the PDF?

**Comment by kylesoskin** [comment]
Reached out to LH to get more info in regards to async nature of their implementation. 

@Mottie it is by submission, not by veteran.

**Comment by saderagsdale** [comment]
@Mottie @kylesoskin, I am unassigning you both and moving this out of the sprint. This is with design right now. Please do not begin coding.

**Comment by saderagsdale** [comment]
Design iteration: What did we decide?
What is the design pattern for this?
- To communicate that the submission/download transaction is happening
- Loading screens are the solution
What are our options?
- Download button
- Send via email
- Download a page that's generated with Veteran's responses (to handle concerns about accessibility)
What are the pros and cons?
- Download button: The button can be shown immediately for accessibility purposes. The backend won't respond right away. Challenges are loading times and definition of submission.
- Send via email: Email comes from Lighthouse already. Need to find out what this says. Could work with VA Notify (informing LH) to determine best way to add this to the email.
- Download a page that's generated with Veteran's responses (to handle concerns about accessibility): Becomes a web page with the response data. Would mean that the version Veterans get doesn't look like the form. Concern is if the Watermark interacts negatively with the form. We don't know if it can be read normally. 
How is it valuable for users?
- It is easier to not have to think about how they'll get the data and wait for loading. Team is tilting toward email as the initial implementation. Veteran can optionally choose. 
How is it valuable for the business?
What did we choose?
- Move forward with email notification, possibly adding web page or another accessible option later
- Action item: Find out what LH is currently sending @Mottie and what state the email current goes out in
- Action item: Consider content changes on confirmation page and work with DK on PDf accessibility requirements @eileen-coforma

**Comment by eileen-coforma** [comment]
Updated **Acceptance Criteria** based on decisions made, as well as design files.
Also added lingering questions to **Open Questions**

**Comment by Mottie** [comment]
[Lighthouse sends an email after the form reaches a "submitted" status](https://dsva.slack.com/archives/CKUCFQAHJ/p1696536474112339?thread_ts=1696520445.573659&cid=CKUCFQAHJ). And they sound open to updating the email to include a link with content to emphasize that it's only available for a limited time; but then Kristen brought up this point: 

> > Quote from Sade: It is crucial that when a Veteran downloads a copy of their claim, they are not left with the impression that their claim was submitted and established if it was not. The purpose of the feature is to increase confidence and give Veterans greater access to their data, and anything less than a confirmed submission hurts that outcome.
>
> we don't really have validation that the appeal was received and entered into Caseflow until the appeal reaches the "complete" status, which can happen days after the "success" status is reached. So if we are building a feature that gives Veterans access to a PDF soon after it's submitted to Lighthouse (whether that's before or after the "submitted" status is reached), we aren't really achieving the goal that Sade mentions above.

**Comment by eileen-coforma** [comment]
@saderagsdale
To Robin's point above: Are we OK with confirming submission at the "submitted" stated, which is when the PDF reaches Central Mail, or do we want to confirm only after the case has been established in Caseflow?

I would be OK with confirmation email and form being sent in "submitted" state, because it is accurate to say VA has confirmed the receipt of the form.

Whether or not the claim gets established is secondary, since it may branch into VA contacting the Veteran to clarify intent or extra information to establish.

**Comment by saderagsdale** [comment]
@eileen-coforma Check with DK on ARIA tag idea

**Comment by eileen-coforma** [comment]
ARIA tag is not required. DK has confirmed with A11y specialists. @saderagsdale designs are ready to go.

**Comment by eileen-coforma** [comment]
@saderagsdale @Mottie I ran the ACs by DK, and caught one mistake. It should be good to go now.
The PDF link can be there, but it does not need to download a PDF for the study.

**Comment by saderagsdale** [comment]
@Mottie Just needs an estimate.

**Comment by rocio-desantiago** [comment]
@eileen-coforma 
Does it make sense to have these questions under the `Issues for Review` header on the `HTML Claim Page`?
```
- Are you requesting an extension?
- Reason for extension
- Are you appealing denial of VA health care benefits
```

**Comment by eileen-coforma** [comment]
@rocio-desantiago 
For this build, yes. Just because that's the way the digital form was originally laid out. If we get feedback from Veterans about the headers not being accurate, it'll be good to capture.

**Comment by HeatherWidmont** [comment]
@rocio-desantiago working on content updates - styling the way its presented on the last page. Should be done today

**Comment by HeatherWidmont** [comment]
@davidakennedy can take a look once Rocio is finished with the content updates

**Comment by davidakennedy** [comment]
I gave this a run through using our [foundational testing standards](https://depo-platform-documentation.scrollhelp.site/collaboration-cycle/prepare-for-an-accessibility-staging-review#Prepareforanaccessibilitystagingreview-Foundationalaccessibilitytests(required)foundational-testing). No major issues on either page.

One minor issue I found on `/decision-reviews/submitted-appeal/`

1. Some paragraphs have the the same id: `id="delay-note"`. This isn't a blocker, and would not interfere with user testing, but would cause problems if we ever referenced that ID in some way. The solution would be to make those IDs unique.

**Comment by Mottie** [comment]
Duplicate ID has been removed. The prototype should now be ready for research.

**Comment by HeatherWidmont** [comment]
Ready for @eileen-coforma to review

**Comment by eileen-coforma** [comment]
@rocio-desantiago 
I just noticed that the date for the confirmation page for submission is dynamically generated to be today's date. 

Can we do that for the header? Board Appeal submitted on January 25, 2024
We want to make sure the information matches between pages limit confusion.

Also, the expiration date for the PDF will also be dynamic because it's 7 days from the date of submission.

**Comment by eileen-coforma** [comment]
The decision dates for contentions are also 2023 instead of 2021.

**Comment by rocio-desantiago** [comment]
This has been merged and should be ready for review by Monday afternoon when it's in staging.

**Comment by HeatherWidmont** [comment]
Team to review in parking lot today
